From: Anthony G. Basile <blueness@gentoo.org>

This patch addresses three issues to make gradm respect aspects
of a typical gentoo system: 1) We do not strip binaries automatically
to permit FEATURES="nostrip" to work, 2) We respect user provided
CFLAGS and LDFLAGS, 3) We install our udev rules in /lib/udev/rules.d.

diff -Naur gradm2.orig//Makefile gradm2/Makefile
--- gradm2.orig//Makefile	2012-07-09 07:20:14.000000000 -0400
+++ gradm2/Makefile	2012-07-09 07:21:44.000000000 -0400
@@ -18,16 +18,16 @@
 BISON=/usr/bin/bison
 #YACC := $(shell if [ -x $(BYACC) ]; then echo $(BYACC); else echo $(BISON); fi)
 YACC=$(BISON)
-MKNOD=/bin/mknod
+MKNOD=true
 #for dietlibc
 #CC=/usr/bin/diet /usr/bin/gcc
 CC=/usr/bin/gcc
 FIND=/usr/bin/find
-STRIP=/usr/bin/strip
+STRIP=true
 LIBS := $(shell if [ "`uname -m`" != "sparc64" -a "`uname -m`" != "x86_64" ]; then echo "-lfl" ; else echo "" ; fi)
 OPT_FLAGS := $(shell if [ "`uname -m`" != "sparc64" ] && [ "`uname -m`" != "x86_64" ]; then echo "-O2" ; else echo "-O2 -m64" ; fi)
-CFLAGS := $(OPT_FLAGS) -Wcast-qual -DGRSEC_DIR=\"$(GRSEC_DIR)\" -D_LARGEFILE64_SOURCE
-LDFLAGS=
+CFLAGS := $(OPT_FLAGS) -Wcast-qual -DGRSEC_DIR=\"$(GRSEC_DIR)\" -D_LARGEFILE64_SOURCE $(CFLAGS)
+LDFLAGS+=
 INSTALL = /usr/bin/install -c
 
 # FHS
@@ -134,10 +134,11 @@
 			$(MKNOD) -m 0622 $(DESTDIR)/dev/grsec c 1 13 ; \
 		fi \
 	fi
-	@if [ -d $(DESTDIR)/etc/udev/rules.d ] ; then \
-		echo "ACTION!=\"add|change\", GOTO=\"permissions_end\"" > $(DESTDIR)/etc/udev/rules.d/80-grsec.rules ; \
-		echo "KERNEL==\"grsec\",          MODE=\"0622\"" >> $(DESTDIR)/etc/udev/rules.d/80-grsec.rules ; \
-		echo "LABEL=\"permissions_end\"" >> $(DESTDIR)/etc/udev/rules.d/80-grsec.rules ; \
+	@mkdir -p $(DESTDIR)/lib/udev/rules.d
+	@if [ -d $(DESTDIR)/lib/udev/rules.d ] ; then \
+		echo "ACTION!=\"add|change\", GOTO=\"permissions_end\"" > $(DESTDIR)/lib/udev/rules.d/80-grsec.rules ; \
+		echo "KERNEL==\"grsec\",          MODE=\"0622\"" >> $(DESTDIR)/lib/udev/rules.d/80-grsec.rules ; \
+		echo "LABEL=\"permissions_end\"" >> $(DESTDIR)/lib/udev/rules.d/80-grsec.rules ; \
 	fi
 	@if [ -f $(DESTDIR)/sbin/udevadm ] ; then \
 		$(DESTDIR)/sbin/udevadm trigger --action=change ; \
